### PR TITLE
Allow tracker to take command line args for a state manager.

### PR DESCRIPTION
**Pull request by cckellogg**
To make deploying easier in a container environment make the config file
optional and allow the traker to be started with command line args.

Args added:
--type; state manager type
--name; state manager name
--rootpath; where state manager data is stored
--hostport; for zookeeper state managers
--tunnelhost; if ssh tunneling is needed

### CODE REVIEW

File: heron/tools/tracker/src/python/main.py [hunk 1]
```diff
@@ -125,13 +124,43 @@ def add_titles(parser):
...
       '--config-file',
-      metavar='(a string; path to config file; default: "' + default_config_file + '")',
-      default=default_config_file)
+      metavar="""(a string; path to config file;)
+      if a config file is provided the following args are ignored:
+      [--type, --name, --rootpath, --tunnelhost, --hostport]
+      """)
+
+  parser.add_argument(
+      '--type',
```
Comment (cckellogg) [inline heron/tools/tracker/src/python/main.py]: Should we prefix these args with 'sm' or something like that?

**Comment by cckellogg** [comment]
@jrcrawfo will you take a look at this when you get the chance?

File: heron/tools/tracker/src/python/main.py (hunk 1 above)
Comment (objmagic) [inline heron/tools/tracker/src/python/main.py]: yeah better to have prefix

File: heron/tools/tracker/src/python/config.py [hunk 2]
```diff
@@ -27,21 +25,11 @@ class Config(object):
...
-
-  def parse_config_file(self, conf_file):
-    """parse config files"""
-    expanded_conf_file_path = os.path.expanduser(conf_file)
-    assert os.path.lexists(expanded_conf_file_path), "Config file does not exists: %s" % (conf_file)
-
-    # Read the configuration file
-    with open(expanded_conf_file_path, 'r') as f:
-      self.configs = yaml.load(f)
+    self.configs = config
```
Comment (objmagic) [inline heron/tools/tracker/src/python/config.py]: there is an `self.configs` above.

File: heron/tools/tracker/src/python/utils.py [hunk 3]
```diff
@@ -150,10 +151,15 @@ def get_heron_tracker_conf_dir():
...
-#   This will provide heron tracker lib directory from .pex file.
-#   :return: absolute path of heron lib directory
-#   """
-#   lib_path = os.path.join(get_heron_tools_dir(), LIB_DIR)
-#   return lib_path
+
+def parse_config_file(config_file):
+  expanded_config_file_path = os.path.expanduser(config_file)
+  assert os.path.lexists(expanded_config_file_path), \
+    "Config file does not exists: %s" % (config_file)
```
Comment (objmagic) [inline heron/tools/tracker/src/python/utils.py]: "exists" -> "exist"

**Comment by cckellogg** [comment]
Putting up a different pull request

**Pull request closed without merging**